src/components/auth/AuthTorus.js

```diff
@@ -0,0 +1,232 @@
+import illustration from '../../assets/Auth/Illustration.svg'
+import config from '../../config/config'
+import { theme as mainTheme } from '../theme/styles'
+import Section from '../common/layout/Section'
+import { getDesignRelativeHeight } from '../../lib/utils/sizes'
+import SimpleStore from '../../lib/undux/SimpleStore'
+import { useErrorDialog } from '../../lib/undux/utils/dialog'
+
+Image.prefetch(illustration)
+const log = logger.child({ from: 'AuthTorus' })
+
+const AuthTorus = ({ screenProps, navigation, styles, store }) => {
+  const asGuest = true
+  const [serviceWorker, setServiceWorker] = useState(undefined)
+  const [showErrorDialog] = useErrorDialog()
+
+  const torus = useMemo(
+    () =>
+      new TorusSdk({
+        typeOfLogin: 'google',
+        verifier: 'google-gooddollar',
+        GOOGLE_CLIENT_ID: config.googleClientId,
+        FACEBOOK_APP_ID: config.facebookAppId,
+        proxyContractAddress: '0x4023d2a0D330bF11426B12C6144Cfb96B7fa6183', // details for test net
+        network: 'ropsten', // details for test net
+        redirect_uri: `${config.publicUrl}/torus/redirect`,
+        enableLogging: config.env === 'development',
+      }),
+    []
+  )
+
+  const registerTorusWorker = async () => {
+    try {
+      const registration = await navigator.serviceWorker.register(`${config.publicUrl}/torus//sw.js`)
+      log.debug('torus service worker registerd', { registration })
+      setServiceWorker(registration)
+    } catch (e) {
+      log.error('failed registering torus service worker', e.message, e)
+    }
+  }
+
+  useEffect(() => {
+    registerTorusWorker()
+    return () => {
+      if (serviceWorker) {
+        serviceWorker.unregister()
+      }
+    }
+  }, [])
+  const goToW3Site = () => {
+    fireEvent(CLICK_BTN_GETINVITED)
+    window.location = config.web3SiteUrl
+  }
+
+  const handleSignUp = async () => {
+    store.set('loadingIndicator')({ loading: true })
+    const redirectTo = 'Phone'
+    let torusUser
+    try {
+      torusUser = await torus.triggerLogin()
+      log.debug('torus login success', { torusUser })
+    } catch (e) {
+      store.set('loadingIndicator')({ loading: false })
+      log.error('torus login failed', e.message, e)
+      showErrorDialog('We were unable to complete the login. Please try again.')
+      return
+    }
+
+    try {
+      if (indexedDB !== undefined) {
+        const req = new Promise((res, rej) => {
+          const del = indexedDB.deleteDatabase('radata')
+          del.onsuccess = res
+          del.onerror = rej
+        })
+        await req
+
+        log.info('indexedDb successfully cleared')
+      }
+    } catch (e) {
+      log.error('Failed to clear indexedDb', e.message, e)
+    } finally {
+      store.set('loadingIndicator')({ loading: false })
+    }
+
+    navigation.navigate(redirectTo, { torusUser })
+
+    //Hack to get keyboard up on mobile need focus from user event such as click
+    setTimeout(() => {
```

Comment:
see the comment above it. its a hack. you cant call focus not from a user action.

Reply:
Yes, on IOS it's prohibited, but we can make an input first responder https://developer.apple.com/documentation/uikit/uiresponder/1621113-becomefirstresponder
It will be auto-focused on first user's interaction

we could add this to icebox for future optimizations